mbof/views.py

```diff
@@ -123,9 +149,37 @@ class IntentionViewSet(OcellusViewSet):
     queryset = Intention.objects.all()
     serializer_class = IntentionSerializer
 
+    def retrieve(self, request, *args, **kwargs):
+        log_action(request)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        serializer_data = serializer.data
+        if serializer_data['respondent'] == self.request.user.username:
```

Comment:
This code is getting repeated a lot.  How about a method like
isRequestByCurrentUser(requestor) returing True or False ?  There would be less code to read and clearer intent.